builder/builder.go

```diff
@@ -466,6 +466,9 @@ func (b *Builder) saveArtifactLocally(ctx context.Context, artifact domain.Artif
 	if err != nil {
 		return errors.Wrapf(err, "glob")
 	} else if !artifact.Target.IsRemote() && len(fromGlobMatches) <= 0 {
+		if ifExists {
```

Comment:
Kept the name `ifExists` everywhere all the way through in an attempt to draw a direct line from the parameter. However; I feel like `optional` might be a better name from the listener on down... open for suggestions if this is offensive.